### DISCUSSION TITLE
Parser object

### DISCUSSION THREAD

**Discussion by Xandaros** [tree-sitter/tree-sitter#1204] [Q&A] [answered]
So I saw one mention that the `Parser` (`TSParser`? Not sure - I use the Rust bindings) object is stateful.

Does that mean I should keep it around for subsequent parses and not just create a new parser?
Should I use separate parsers for different documents, or would it be find to reuse the same one?

**Comment by ahlinc** [comment, marked as answer]
You can reuse the same `Parser` object as many times as you need and you need additional `Parser` objects only if you want to have multiple parsings that happen in parallel on multiple threads.
